File: manager_frontend/src/features/documents/model/transport-terms.ts
export type TransportTerms = {
  car_model: string | null;
  car_number: string | null;
  driver_name: string | null;
  carrier: string | null;
};

export const createDefaultTransportTerms = (): TransportTerms => ({
  car_model: null,
  car_number: null,
  driver_name: null,
  carrier: null,
});

export const serializeTransportTerms = (terms: TransportTerms): TransportTerms => ({
  car_model: terms.car_model?.trim() || null,
  car_number: terms.car_number?.trim() || null,
  driver_name: terms.driver_name?.trim() || null,
  carrier: terms.carrier?.trim() || null,
});

File: modules/documents/domain/transport_terms.py
from __future__ import annotations

from dataclasses import dataclass


WAYBILL_DOCUMENT_TYPES = frozenset({"tn2", "ttn1"})


@dataclass(frozen=True, slots=True)
class TransportTerms:
    """Transport facts frozen into a native paper waybill draft."""

    car_model: str | None = None
    car_number: str | None = None
    driver_name: str | None = None
    carrier: str | None = None

